static-analysis/lint/src/main/kotlin/com/uber/autodispose/lint/AutoDisposeDetector.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2018. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.lint
+
+import com.android.tools.lint.client.api.JavaEvaluator
+import com.android.tools.lint.detector.api.*
+import com.intellij.psi.PsiMethod
+import org.jetbrains.uast.UCallExpression
+import java.util.EnumSet
+
+/**
+ * Detector which checks if your stream subscriptions are handled by AutoDispose.
+ */
+class AutoDisposeDetector: Detector(), SourceCodeScanner {
+
+  companion object {
+    val ISSUE: Issue = Issue.create(
+        "AutoDisposeUsage",
+        "Subscription not managed with AutoDispose",
+        "You're subscribing to an observable but not handling it's subscription. This " +
```

Comment:
I'm not too happy with the messages here so I'm open to suggestions!

Reply:
Can we borrow the messaging from the error prone checker?